src/plugins/intel_cpu/src/nodes/strided_slice.cpp

```diff
@@ -38,8 +40,110 @@ bool StridedSlice::isSupportedOperation(const std::shared_ptr<const ov::Node>& o
     return true;
 }
 
+class StridedSliceShapeInfer : public ShapeInferEmptyPads {
+public:
+    StridedSliceShapeInfer(const IShapeInfer::port_mask_t& port_mask,
+                           size_t output_size,
+                           std::unordered_set<int64_t> begin_mask,
+                           std::unordered_set<int64_t> end_mask,
+                           std::unordered_set<int64_t> new_axis_mask,
+                           std::unordered_set<int64_t> shrink_axis_mask)
+    : m_port_mask(port_mask), m_outputShape(output_size, 1),
+      m_begin_mask_set(std::move(begin_mask)),
+      m_end_mask_set(std::move(end_mask)),
+      m_new_axis_mask_set(std::move(new_axis_mask)),
+      m_shrink_axis_mask_set(std::move(shrink_axis_mask)) {}
+
+    Result infer(
+        const std::vector<std::reference_wrapper<const VectorDims>>& input_shapes,
+        const std::unordered_map<size_t, MemoryPtr>& data_dependency) override {
+        const VectorDims& shapeIn = input_shapes[0].get();
+        const VectorDims& shapeBegin = input_shapes[1].get();
+        if (data_dependency.at(1)->GetDataType() != dnnl::memory::data_type::s32) {
+            IE_THROW(Unexpected) << "The data type of begin/end/stride is NOT s32, which is unexpected!";
+        }
+        auto beginPtr = reinterpret_cast<int32_t *>(data_dependency.at(1)->GetPtr());
+        auto endPtr = reinterpret_cast<int32_t *>(data_dependency.at(2)->GetPtr());
+        auto stridePtr = reinterpret_cast<int32_t *>(data_dependency.at(3)->GetPtr());
+
+        for (auto i = 0, new_idx = 0; i < shapeIn.size(); ++i) {
+            if (m_new_axis_mask_set.count(i)) {
+                // deal with new_axis_mask
+                m_outputShape[new_idx] = 1;
+                m_outputShape[new_idx+1] = shapeIn[i];
+                new_idx+=2;
+            } else if (m_shrink_axis_mask_set.count(i)) {
+                // deal with shrink_axis_mask
```

Comment:
If this is something we need to do later, plaese add TODO statement to the comment. But how does it work now?

Reply:
> If this is something we need to do later, plaese add TODO statement to the comment. But how does it work now?

There is nothing to do if the `shrink_axis` is `true`. Just go head without doing anything.

Reply:
Then it is better to replace the last else with `else if (!m_shrink_axis_mask_set.count(i))` since this condition is checked anyway, and remove this line.

Reply:
> Then it is better to replace the last else statement with `else if (!m_shrink_axis_mask_set.count(i))` since this condition is checked anyway, and remove this line with the comment since the purpose will be clear from the code.

sure. re-arrange these codes.